From 7c9d61957f23b8c935b88a60b89779fc7faabb2c Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 29 Nov 2016 19:26:01 -0800
Subject: [PATCH 0579/1239] Fixed order of parameters to writeq, fixes GPIO not
 working right

The order the parameters was passed to writeq was reversed.

Also fixed a compiler warning.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 drivers/gpio/thunderx_gpio.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpio/thunderx_gpio.c b/drivers/gpio/thunderx_gpio.c
index 8cf0a76327..2237e96686 100644
--- a/drivers/gpio/thunderx_gpio.c
+++ b/drivers/gpio/thunderx_gpio.c
@@ -62,7 +62,7 @@ static int octeontx_gpio_dir_input(struct udevice *dev, unsigned offset)
 	struct octeontx_gpio *gpio = dev_get_priv(dev);
 	debug("%s(%s, %u)\n", __func__, dev->name, offset);
 	clrbits_le64(gpio->baseaddr + GPIO_BIT_CFG(offset),
-		     (0x3ffUL << 16) | 1UL);
+		     (0x3ffUL << 16) | 4UL | 1UL);
 	__iowmb();
 
 	return 0;
@@ -74,10 +74,10 @@ static int octeontx_gpio_dir_output(struct udevice *dev, unsigned offset,
 	struct octeontx_gpio *gpio = dev_get_priv(dev);
 
 	debug("%s(%s, %u, %d)\n", __func__, dev->name, offset, value);
+	writeq(GPIO_BIT(offset), gpio->baseaddr + GPIO_TX_REG(offset, value));
 
-	writeq(gpio->baseaddr + GPIO_TX_REG(offset, value), GPIO_BIT(offset));
 	clrsetbits_le64(gpio->baseaddr + GPIO_BIT_CFG(offset),
-			(0x3ffUL << 16), 1UL);
+			((0x3ffUL << 16) | 4UL), 1UL);
 	__iowmb();
 
 	return 0;
@@ -101,7 +101,7 @@ static int octeontx_gpio_set_value(struct udevice *dev,
 	struct octeontx_gpio *gpio = dev_get_priv(dev);
 
 	debug("%s(%s, %u, %d)\n", __func__, dev->name, offset, value);
-	writeq(gpio->baseaddr + GPIO_TX_REG(offset, value), GPIO_BIT(offset));
+	writeq(GPIO_BIT(offset), gpio->baseaddr + GPIO_TX_REG(offset, value));
 
 	return 0;
 
@@ -112,7 +112,7 @@ static int octeontx_gpio_get_function(struct udevice *dev,
 {
 	struct octeontx_gpio *gpio = dev_get_priv(dev);
 	u64 pinsel = readl(gpio->baseaddr + GPIO_BIT_CFG(offset));
-	debug("%s(%s, %u): pinsel: 0x%lx\n", __func__, dev->name, offset,
+	debug("%s(%s, %u): pinsel: 0x%llx\n", __func__, dev->name, offset,
 	      pinsel);
 	if (GPIO_BIT_CFG_FN(pinsel))
 		return GPIOF_FUNC;
-- 
2.29.0

